src/test/app/AccountDelete_test.cpp

```diff
@@ -912,6 +912,301 @@ class AccountDelete_test : public beast::unit_test::suite
+
+            Env env{*this};
+            env.fund(XRP(100000), alice, becky, carol);
+            env.close();
+
+            // carol issue credentials for becky
+            env(credentials::create(becky, carol, credType));
+            env.close();
+            env(credentials::accept(becky, carol, credType));
+            env.close();
+
+            // get credentials index
+            auto const jv =
+                credentials::ledgerEntry(env, becky, carol, credType);
+            std::string const credIdx = jv[jss::result][jss::index].asString();
+
+            // Close enough ledgers to be able to delete carol's account.
+            incLgrSeqForAccDel(env, carol);
+
+            auto const acctDelFee{drops(env.current()->fees().increment)};
+            env(acctdelete(carol, alice), fee(acctDelFee));
+            env.close();
+
+            {  // check that credential object deleted too
+                BEAST_EXPECT(!env.le(credIdx));
+                auto const jv =
+                    credentials::ledgerEntry(env, becky, carol, credType);
+                BEAST_EXPECT(
+                    jv.isObject() && jv.isMember(jss::result) &&
+                    jv[jss::result].isMember(jss::error) &&
+                    jv[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+
+        {
+            testcase("Deleting Subject deletes issued credentials");
+
+            using namespace test::jtx;
+
+            Account const alice{"alice"};
+            Account const becky{"becky"};
+            Account const carol{"carol"};
+
+            const char credType[] = "abcd";
+
+            Env env{*this};
+            env.fund(XRP(100000), alice, becky, carol);
+            env.close();
+
+            // carol issue credentials for becky
+            env(credentials::create(becky, carol, credType));
+            env.close();
+            env(credentials::accept(becky, carol, credType));
+            env.close();
+
+            // get credentials index
+            auto const jv =
+                credentials::ledgerEntry(env, becky, carol, credType);
+            std::string const credIdx = jv[jss::result][jss::index].asString();
+
+            // Close enough ledgers to be able to delete carol's account.
+            incLgrSeqForAccDel(env, becky);
+
+            auto const acctDelFee{drops(env.current()->fees().increment)};
+            env(acctdelete(becky, alice), fee(acctDelFee));
+            env.close();
+
+            {  // check that credential object deleted too
+                BEAST_EXPECT(!env.le(credIdx));
+                auto const jv =
+                    credentials::ledgerEntry(env, becky, carol, credType);
+                BEAST_EXPECT(
+                    jv.isObject() && jv.isMember(jss::result) &&
+                    jv[jss::result].isMember(jss::error) &&
+                    jv[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+    }
```

Comment:
There should be additional tests for subject and issuer deletion for unaccepted credentials too. Could be added to the same tests with an additional credential on each.

Reply:
Please check testCredentialsDelete() in Credentials_test.cpp. They are there cause we check deletion of the credential object, not the deleteAccount itself.

Reply:
Why have the tests in two places? It's a bit confusing.

Reply:
There are some overlapped tests as this functionality overlap many transactions. But they are slightly different in their own way